arrow/arrow.py

```diff
@@ -628,6 +661,10 @@ def replace(self, **kwargs):
 
             if key in self._ATTRS:
                 absolute_kwargs[key] = value
+            elif key == "fold":
+                # datetime replace will not work in 2.7/3.5
+                current = dateutil_tz.enfold(self._datetime, fold=value)
+                return self.fromdatetime(current)
```

Comment:
Couldn't this cause issues if a user passes in an absolute week or quarter (e.g. going around the key check below)?

Also, should we check for <Python 3.6 and only return if Python 2.7 or Python 3.5?

Reply:
```shell
(arrow27) chris@ThinkPad:~/arrow$ python
Python 2.7.18 (tags/v2.7.18:8d21aa21f2, Jul 16 2020, 23:20:47) 
[GCC 9.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
arrow/arrow.py:28: DeprecationWarning: Arrow will drop support for Python 2.7 and 3.5 in the upcoming v1.0.0 release. Please upgrade to Python 3.6+ to continue receiving updates for Arrow.
  DeprecationWarning,
>>> dt=arrow.utcnow()
>>> dt
<Arrow [2020-08-17T15:41:34.465825+00:00]>
>>> dt.replace(quarter=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "arrow/arrow.py", line 670, in replace
    raise AttributeError("setting absolute {} is not supported".format(key))
AttributeError: setting absolute quarter is not supported
>>> dt.replace(fold=1, quarter=1)
<Arrow [2020-08-17T15:41:34.465825+00:00]>
```
This is what happens currently, no change but it does get past the check.

We really need to use enfold here for consistency on all versions.

Reply:
Oh ok, but this will throw an error if we do `dt.replace(quarter=1, fold=1)`. Do you think this could be a bit misleading if a user writes `dt.replace(fold=1, quarter=1)` because they may think that the quarter is being set when it really is not?

The bigger problem here lies with the fact that replace won't work if arguments are passed beyond fold the fold argument:

```python
>>> dt=arrow.utcnow()
>>> dt
<Arrow [2020-08-17T18:44:46.718769+00:00]>
>>> dt.replace(day=19)
<Arrow [2020-08-19T18:44:46.718769+00:00]>
>>> dt.replace(fold=1, day=19)
<Arrow [2020-08-17T18:44:46.718769+00:00]>
```

This bug occurs because you are returning rather than continuing the loop.

Reply:
Ah yeah you're completely right, good spot. Let me think about how to fix this.